aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+    <textarea data-val="true" 
+     data-val-maxlength="The field Description must be a string or array type with a maximum length of &#x27;1024&#x27;."
+     data-val-maxlength-max="1024" 
+     data-val-minlength="The field Description must be a string or array type with a minimum length of &#x27;5&#x27;." 
+     data-val-minlength-min="5" 
+     id="Description" name="Description">
+    </textarea>
+    <button type="submit">Test</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+The Label Tag Helper
+--------------------
+
+- Generates the label caption and ``for`` attribute on a `<label> <https://www.w3.org/wiki/HTML/Elements/label>`__ element for an expression name
+- HTML Helper alternative: ``Html.LabelFor``.
+
+The `Label Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/LabelTagHelper/index.html>`__  provides the following benefits over a pure HTML label element:
+
+- You automatically get the descriptive label value from the ``Display`` attribute. The intended display name might change over time, and the combination of ``Display`` attribute and Label Tag Helper will apply the ``Display`` everywhere it's used.
+- Less markup in source code
+- Strong typing with the model property.
+
+Sample:
+
+.. literalinclude::  forms/sample/final/ViewModels/SimpleViewModel.cs
+  :language: c#
+
+..  literalinclude::  forms/sample/final/Views/Demo/RegisterLabel.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+
+The following HTML is generated for the ``<label>`` element:
+
+.. code-block:: HTML
+
+ <label for="Email">Email Address</label>  
+ 
+The Label Tag Helper generated the ``for`` attribute value of "Email", which is the ID associated with the ``<input>`` element. The Tag Helpers generate consistent ``id`` and ``for`` elements so they can be correctly associated. The caption in this sample comes from the ``Display`` attribute. If the model didn't contain a ``Display`` attribute, the caption would be the property name of the expression.
+ 
+The Validation Tag Helpers
+---------------------------
+
+There are two Validation Tag Helpers. The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__ (which displays a validation message for a single property on your model), and the `Validation Summary Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationSummaryTagHelper/index.html>`__ (which displays a summary of validation errors). The `Input Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/InputTagHelper/index.html>`__ adds HTML5 client side validation attributes to input elements based on data annotation attributes on your model classes. The Validation Tag Helper displays these error messages when a validation error occurs. 
```

Comment:
Odd to mention only "client side validation" here and in the bullets below.

Background: The HTML is generated with any server-side error messages listed. Then JavaScript code clears errors and adds new ones as the user changes values. Since some validations can only be done on the server, client-side validations aren't necessarily complete.